src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -135,7 +135,15 @@ public void End(ProduceEndType endType)
 
         private void ScheduleWrite()
         {
-            _thread.Post(_this => _this.WriteAllPending(), this);
+            // Don't post write to closed socket
+            if (!_socket.IsClosed)
```

Comment:
I think its only in a shutdown race situation; happens in both windows and mono. When it happens windows seems ok with it, but mono very unhappy.